### PR TITLE
Move selection commands to command module

### PR DESCRIPTION
**Pull request by gchp**
Is this what you were thinking?

### CODE REVIEW

**Comment by kisielk** [comment]
Yep, that's much better. The modes should consist mostly of commands being sent in to the commands channel and not much else.

File: commands/selection.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
+package commands
+
+import (
+	"github.com/kisielk/vigo/editor"
+)
+
+const (
+	Up   Dir = 3
```
Comment (kisielk) [inline commands/selection.go]: Put these with Forward and Backward

**Comment by gchp** [comment]
Updated from your comments.

Just noticed an issue with the selection not clearing when existing visual mode. Instead of just adding the FIXME again for this, it might be better to just implement the `editor.Redraw()` method. It seems like there is a redraw channel created on the editor, which I assume is for this purpose?

**Comment by kisielk** [comment]
Redraw won't do anything unless the dirty flag is set on the view. I'll see if I can fix that.

**Pull request merged**